metaserver/src/main/scala/scala/meta/languageserver/storage/Bytes.scala

```diff
@@ -0,0 +1,35 @@
+package scala.meta.languageserver.storage
+
+import java.nio.charset.StandardCharsets
+import org.langmeta.internal.semanticdb.schema.Database
+import org.langmeta.io.AbsolutePath
+
+trait FromBytes[A] { self =>
+  def fromBytes(bytes: Array[Byte]): A
+  def map[B](f: A => B): FromBytes[B] =
+    bytes => f(self.fromBytes(bytes))
+}
+object FromBytes {
+  implicit val StringFromBytes: FromBytes[String] =
+    new String(_, StandardCharsets.UTF_8)
+  implicit val ByteArrayFromBytes: FromBytes[Array[Byte]] =
+    identity[Array[Byte]]
+  implicit val DatabaseFromBytes: FromBytes[Database] =
+    bytes => Database.parseFrom(bytes)
+}
+
+trait ToBytes[A] { self =>
+  def toBytes(e: A): Array[Byte]
+  def map[B](f: B => A): ToBytes[B] =
```

Comment:
I don't understand why this is B => A instead of A => B.

Reply:
Renamed to `contramap` https://typelevel.org/cats/typeclasses/contravariant.html